packages/core/src/transport/httpRequest.ts

```diff
@@ -21,6 +22,8 @@ export class HttpRequest {
         if (isQueued) {
           return
         }
+
+        addFailedSendBeacon(this.endpointBuilder.endpointType, size, reason)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**❓ question:**  ‏how do we plan to differentiate collisions from other possible failure sources?
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->